src/transaction/components/TransactionProvider.tsx

```diff
@@ -82,12 +105,77 @@ export function TransactionProvider({
     setLifeCycleStatus,
     transactionHashList,
   });
-  const { transactionHash, status: callStatus } = useCallsStatus({
+  /*
+    useSendCalls or useSendCall
+    Used for contract calls with raw calldata.
+  */
+  const { status: statusSendCalls, sendCallsAsync } = useSendCalls({
     setLifeCycleStatus,
-    transactionId,
+    setTransactionId,
+  });
+  const {
+    status: statusSendCall,
+    sendCallAsync,
+    data: sendCallTransactionHash,
+  } = useSendCall({
+    setLifeCycleStatus,
+    transactionHashList,
+  });
+
+  /* 
+    Transactions
+    Can be of type call or contracts.
+  */
+  const [transactions, setTransactions] = useState<
```

Comment:
Move this up, in the `// Core Hooks` section.

We want our Hooks organized by the
- Core one that not depends from others
- The hooks that depand from others.